src/context/videoContext.js

```diff
@@ -55,8 +74,45 @@ const VideoProvider = ({children}) => {
         }
         allCategories();
     }, []);
+
+    const getHistory = async ( token, video ) =>{
+        if(token){
+        try{
+            const response = await postHistoryVideos( token, video )
+            videoDispatch({ type : "ADD_HISTORY", payload : response.history})  
+        }
+        catch(error){
+            console.log(error)
+        }
+    }
+    else {
+        navigate("/login")
+    }
+    }
+
+    const removeHistory = async ( token, _id) =>{
+        try{
+            const response = await removeHistoryVideos(token, _id)
+            videoDispatch({type : "REMOVE_HISTORY", payload : response.history})
+        }
+        catch(error){
+            console.log(error)
```

Comment:
try using toast, do not keep console.logs

Reply:
Oh okay, I have yet to learn how react toast works but will change consoles with them for sure.